ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -176,10 +191,16 @@ class JITModBuilder {
                   CF->getName(), JITMod);
               VMap[CF] = DeclFunc;
             }
+            copyInstruction(&Builder, (Instruction *)&*I);
             // We should expect an "unmappable hole" in the trace. This is
             // where the trace followed a call into external code for which be
             // have no IR, and thus we cannot map blocks for.
             ExpectUnmappable = true;
+            // Peek at the instruction we will need to resume at.
+            auto PeekIter = I;
```

Comment:
Aren't these three lines the same as `ExcCallInst = &*(I++);` ?

Reply:
`PeekIter = I` makes a copy of the iterator, leaving the original iterator untouched. Apparently this is how you peek an iterator in C++: copy the iterator.

Reply:
My eyes hurt! But I have no idea about this stuff, so I shall not rant further :)

Reply:
It's a C++ thing. When you have a value on the stack, and assign something from it, you call a "copy constructor" to make a fresh instance.

It tripped me too when I first started with C++.

Reply:
Just to be 100% sure:

```
#include <iostream>
#include <vector>

using namespace std;

int main(void) {
    vector<int> MyVec = {1, 2, 3};
    auto it1 = MyVec.begin();
    auto it2 = it1;
    it2 += 2;
    cout << "it1: " << *it1 << "\n";
    cout << "it2: " << *it2 << "\n";
}
```

Gives:
```
it1: 1
it2: 3
```